packages/miniflare/src/index.ts

```diff
@@ -1343,19 +1343,41 @@ export class Miniflare {
 					directSocket.host,
 					directSocket.port
 				);
-				sockets.push({
-					name,
-					address,
-					service: {
-						name: getUserServiceName(workerName),
-						entrypoint: entrypoint === "default" ? undefined : entrypoint,
-					},
-					http: {
-						style: directSocket.proxy ? HttpOptions_Style.PROXY : undefined,
-						cfBlobHeader: CoreHeaders.CF_BLOB,
-						capnpConnectHost: HOST_CAPNP_CONNECT,
-					},
-				});
+
+				// only if workers with assets with default exports
+				// (class or non-class based)
+				if (
+					this.#sharedOpts.core.unsafeEnableAssetsRpc &&
+					workerOpts.assets.assets &&
+					entrypoint === "default"
+				) {
+					sockets.push({
+						name,
+						address,
+						service: {
+							name: `${RPC_PROXY_SERVICE_NAME}:${workerOpts.core.name}`,
+						},
```

Comment:
is the service the only thing that changes in the if vs else brackets? if so, wouldn't it be clearer to keep the call `sockets.push` the exact same but only conditionally swap services? something like this:
![Screenshot 2025-03-19 at 15 48 58](https://github.com/user-attachments/assets/c7354bd4-8d86-420f-a983-80d5b1aee55e)
or
![Screenshot 2025-03-19 at 15 51 24](https://github.com/user-attachments/assets/a51c585e-d132-4e36-a1dc-0a6b9cd9eadb)

Reply:
yeah sure, I can abs do that!